### PR TITLE
feat(core): Logout should invalidate the auth token (no-changelog)

### PR DESCRIPTION
**Pull request by netroy**
## Summary
When a user logs out, we should also invalidate the auth token to prevent it from being used again.

## Related Linear tickets, Github issues, and Community forum posts
https://linear.app/n8n/issue/SEC-75

## Review / Merge checklist

- [x] PR title and summary are descriptive

### CODE REVIEW

**Review by alexgrozav — DISMISSED** [review]

File: packages/cli/src/databases/migrations/common/1723627610222-CreateInvalidAuthTokenTable.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import type { MigrationContext, ReversibleMigration } from '@db/types';
+
+const tableName = 'invalid_auth_token';
+
+export class CreateInvalidAuthTokenTable1723627610222 implements ReversibleMigration {
+	async up({ schemaBuilder: { createTable, column } }: MigrationContext) {
+		await createTable(tableName)
+			.withColumns(column('token').varchar(512).primary, column('expiresAt').timestamp().notNull)
+			.withIndexOn('token', true);
```
Comment (tomi) [inline packages/cli/src/databases/migrations/common/1723627610222-CreateInvalidAuthTokenTable.ts:9]: Both sqlite and postgres create a unique index for every PK automatically, so this creates a duplicate index. No idea about mysql.

Sqlite:
image


Postgres:
image

File: packages/cli/test/integration/auth.api.test.ts [hunk 2]
```diff
@@ -386,13 +386,19 @@ describe('GET /resolve-signup-token', () => {
...
+		const response = await ownerAgent.post('/logout');
 
 		expect(response.statusCode).toBe(200);
 		expect(response.body).toEqual(LOGGED_OUT_RESPONSE_BODY);
 
 		const authToken = utils.getAuthToken(response);
 		expect(authToken).toBeUndefined();
+
+		ownerAgent.jar.setCookie(`${AUTH_COOKIE_NAME}=${cookie!.value}`);
+		await ownerAgent.get('/login').expect(401);
```
Comment (tomi) [inline packages/cli/test/integration/auth.api.test.ts:402]: Should this be its own test case? Also should we try some other endpoint than `/login` as it's not normally used with the auth token?

**Review by tomi — COMMENTED** [review]
Functionality and approach seem decent 👌 Couple comments about implementation details

File: packages/cli/test/integration/auth.api.test.ts (hunk 2 above)
Comment (netroy) [reply packages/cli/test/integration/auth.api.test.ts:402]: we should rename `GET /login` to `GET /me` instead. it keeps confusing people.

**Review by tomi — APPROVED** [review]
🚀

**Comment by janober** [comment]
Got released with [`n8n@1.57.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.57.0)

**Pull request merged**